includes/modules/category_row.php

```diff
@@ -14,40 +14,93 @@
     die('Illegal Access');
 }
 
-$title = '';
 $num_categories = $categories->RecordCount();
 
-$row = 0;
-$col = 0;
+$rows = 0;
+$columns = 0;
 $list_box_contents = [];
-if ($num_categories > 0) {
-    if ($num_categories < MAX_DISPLAY_CATEGORIES_PER_ROW || MAX_DISPLAY_CATEGORIES_PER_ROW === 0) {
-        $col_width = floor(100/$num_categories);
-    } else {
-        $col_width = floor(100/MAX_DISPLAY_CATEGORIES_PER_ROW);
+$title = '';
+
+if (empty($num_categories)) {
+    return;
+}
+
+$columns_per_row = defined('MAX_DISPLAY_CATEGORIES_PER_ROW') ? MAX_DISPLAY_CATEGORIES_PER_ROW : 0;
+$product_listing_layout_style = (int)$columns_per_row > 1 ? 'columns' : 'fluid';
+if ($columns_per_row === 'fluid') $product_listing_layout_style = 'fluid';
```

Comment:
Should this be

if ($product_listing_layout_style === 'fluid') 
?   Because $columns_per_row is integer.

Reply:
`$columns_per_row` is a string, which comes from the admin config setting. Yes, it should be numeric. And `'0'` means "fluid".
Originally I was accounting for the possibility of someone typing "fluid" instead of typing '0'.

However, I'd forgotten that we actually do have some numeric-only validation on that config setting.
I'll remove it.